pkg/standalone-utils/contracts/ProtocolFeeSplitter.sol

```diff
@@ -0,0 +1,152 @@
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/IProtocolFeeSplitter.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/Authentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/VaultHelpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IProtocolFeesCollector.sol";
+
+interface Pool {
+    function getOwner() external returns (address);
+}
+
+// Packed to use 1 storage slot
+// 1e18 (100% - maximum fee value) can fit in uint96
+struct RevenueShareSettings {
+    uint96 revenueSharePercentageOverride;
+    address beneficiary;
+}
+
+/**
+ * @dev This contract is responsible for splitting the BPT profits collected
+ * by ProtocolFeeCollector between the pool's owner and Balancers DAO treasury
+ * Nothing happens for non-BPT tokens (WETH, WBTC, etc...)
+ */
+contract ProtocolFeeSplitter is IProtocolFeeSplitter, Authentication {
+    using Math for uint256;
+
+    // All fee percentages are 18-decimal fixed point numbers.
+    // Absolute maximum fee percentages (1e18 = 100%, 1e16 = 1%).
+    uint256 private constant _MIN_REVENUE_SHARING_FEE_PERCENTAGE = 1e16; // 1%
+    uint256 private constant _MAX_REVENUE_SHARING_FEE_PERCENTAGE = 50e16; // 50%
+    address private constant _DELEGATE_OWNER = 0xBA1BA1ba1BA1bA1bA1Ba1BA1ba1BA1bA1ba1ba1B;
+
+    IProtocolFeesCollector public immutable protocolFeesCollector;
+
+    // Balancer DAO Multisig
+    address public immutable treasury;
+
+    // Can be updated by BAL governance (1e18 = 100%, 1e16 = 1%).
+    uint256 public defaultRevenueSharingFeePercentage;
+
+    // poolId => PoolSettings
+    mapping(bytes32 => RevenueShareSettings) public poolSettings;
+
+    constructor(IProtocolFeesCollector _protocolFeesCollector, address _treasury)
+        // The ProtocolFeeSplitter is a singleton, so it simply uses its own address to disambiguate action
+        // identifiers.
+        Authentication(bytes32(uint256(address(this))))
+    {
+        protocolFeesCollector = _protocolFeesCollector;
+        treasury = _treasury;
+    }
+
+    function setRevenueSharingFeePercentage(bytes32 poolId, uint256 newSwapFeePercentage)
+        external
+        override
+        authenticate
+    {
+        _require(newSwapFeePercentage >= _MIN_REVENUE_SHARING_FEE_PERCENTAGE, Errors.SPLITTER_FEE_PERCENTAGE_TOO_LOW);
+        _require(newSwapFeePercentage <= _MAX_REVENUE_SHARING_FEE_PERCENTAGE, Errors.SPLITTER_FEE_PERCENTAGE_TOO_HIGH);
+        poolSettings[poolId].revenueSharePercentageOverride = uint96(newSwapFeePercentage);
+        emit PoolRevenueShareChanged(poolId, newSwapFeePercentage);
+    }
+
+    function setDefaultRevenueSharingFeePercentage(uint256 feePercentage) external override authenticate {
+        _require(feePercentage <= _MAX_REVENUE_SHARING_FEE_PERCENTAGE, Errors.SPLITTER_FEE_PERCENTAGE_TOO_HIGH);
```

Comment:
Do we not need second check. Cant they technically still make it 0?

Reply:
Not really, maybe this max check is not necessary as well. If Balancer's DAO wants to have it at some value, then i guess it should be that way